### DISCUSSION TITLE
Is there a Query Builder for form submissions ?

### DISCUSSION THREAD

**Discussion by QuentinGab** [statamic/cms#6446] [Q&A] [answered]
Hi,

I wonder if there is a way to query specific form submissions with a Query Builder (all submissions from a user with a hidden user field in the form blueprint for example). I dug the code a little but not found anything like a Query Builder for Form submissions.

**Comment by jasonvarga** [comment, marked as answer]
There is no query builder for submissions at the moment.

For now, you'll need to get all the submissions then do a regular collection based filter.

```php
$form->submissions()->filter(function ($submission) {
  return $submission->field === 'something';
});
```

**Reply by QuentinGab** [reply]
Ok, thanks for the super fast answer !

**Reply by ryanmitchell** [reply]
@jasonvarga do you think theres any merit in a Submissions query builder? This has been asked a few times now and I'd be happy to work on it if you think its worth it.

**Reply by jasonvarga** [reply]
It's not a simple thing to add, since you gotta start storing stuff in the stache (and then in the DB) but it wouldn't hurt to have it.

**Reply by QuentinGab** [reply]
Just to add some context and why a query builder for submissions would have been awesome for me. 
I'm building an app where user can register to event and each event can have its own specific form to be able to choose different options, consent to specific optin ... I intended to take advantage of the awesome Statamic form builder and use submissions as registration entry.

But in this context, whitout query builder, I would have no other choices than to query all submissions from all form just to be able to tell to the user in which event he has a registration. Not ideal. And as I want to stay flat, I will certainly have to create a collection to use it as a relationship pivot between an event/user and a form submission

**Reply by ryanmitchell** [reply]
I worked up a PR for this here: https://github.com/statamic/cms/pull/6455
